File: CodeProjects/GroundControlStation/GroundControlStation/GroundControlStation/Messages/SyncMessage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GroundControlStation.Messages
{
    public class SyncMessage: Message
    {
        public const byte MessageType = 3;

        public byte RequestedMessage;

        /// <summary>
        /// This represents the number of bytes in this message including the message type.
        /// </summary>
        public const int NumOfBytesInMsg = sizeof(byte) + sizeof(byte);

        public SyncMessage() 
            : base(MessageType, NumOfBytesInMsg)
        {
        }

        public static SyncMessage BuildMessageSt(byte[] byteBuffer)
        {
            SyncMessage sync = new SyncMessage();
            sync.BuildMessage(byteBuffer);

            return sync;
        }

        public override void BuildMessage(byte[] byteBuffer)
        {
            this.MsgType = byteBuffer[0];
            this.RequestedMessage = byteBuffer[1];
        }

        public override byte[] GetRawBytes()
        {
            byte[] rawMsg = new byte[NumOfBytesInMsg];

            rawMsg[0] = MsgType;
            rawMsg[1] = RequestedMessage;

            return rawMsg;
        }
    }
}

File: CodeProjects/FlightComputer/Helicopter/Helicopter/messages/SyncMessage.cpp
/*
 * SyncMessage.cpp
 *
 * Created: 9/15/2013 2:37:16 PM
 *  Author: HP User
 */ 

#include "SyncMessage.h"

#include <string.h>

using namespace helicopter::messages;


byte *SyncMessage::getBytes()
{
	byte *msg = new byte[MessageSize];
	byte *msgPtr = msg;
	msgPtr[0] = this->msgType;
	msgPtr[1] = this->RequestedMessage;
	
	return msg;
}

void SyncMessage::buildMessage(byte *message)
{
	if (message != NULL)
	{
		msgType = message[0];
		RequestedMessage = message[1];
	}
}

SyncMessage* SyncMessage::buildMessageSt(byte *message)
{
	SyncMessage *msg = new SyncMessage(0);
	msg->buildMessage(message);
	
	return msg;
}